android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -338,7 +349,8 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             @Override
             public void afterTextChanged(Editable s) {
                 doneButton.setImageResource(R.drawable.ic_done_grey);
-                addOnDoneAction(referenceEditText, doneButton, widgetArgs);
+                if (widgetArgs.getJsonObject().optBoolean(JsonFormConstants.EXPAND_ON_TEXT_CHANGE))
```

Comment:
Could this moved into this if block https://github.com/OpenSRP/opensrp-client-native-form/pull/509/files#diff-5faa6441c13d21a9d335d1851497f620c839a7c50ec84a6b1b77b96892e94d1aL344.

Unless the entry is valid, we should not generate repeating groups.